dm/pkg/binlog/position.go

```diff
@@ -398,37 +411,23 @@ func (l *Location) ResetSuffix() {
 	l.Suffix = 0
 }
 
-// SetGTID set new gtid for location
-// Use this func instead of GITSet.Set to avoid change other location.
+// SetGTID set new gtid for location.
+// TODO: don't change old Location and return a new one to copy-on-write.
```

Comment:
>  don't change old Location and return a new one to copy-on-write

I can't remember why we need to do this, can you please give me some tips? is this about the func`location.CloneWithFlavor` ?

Reply:
it's a new request. mainly about reducing the possibility that when two variables hold the same reference to a location, changing one of them will accidentally change the other one: SetGTID should not change the input location. I think new API is less error-prone

this problem has another solutions, for example, we use location struct not location pointer, so if a location is assigned to another variables they are different copies. we can choose one solution in following pr.